usaspending_api/awards/management/commands/generate_unlinked_awards_download.py

```diff
@@ -0,0 +1,218 @@
+from usaspending_api.awards.management.sql.spark.unlinked_assistance_file_d2 import file_d2_sql_string
+from usaspending_api.awards.management.sql.spark.unlinked_accounts_file_c import file_c_sql_string
+from usaspending_api.download.filestreaming.file_description import build_file_description, save_file_description
+from usaspending_api.download.filestreaming.zip_file import append_files_to_zip_file
+from usaspending_api.references.models.toptier_agency import ToptierAgency
+
+
+class ComputeTypeEnum(Enum):
+    SPARK = "spark"
+
+
+logger = logging.getLogger("script")
+
+# This list contains any abnormal characters in agency names
+# This list is important to track which characters we need to replace in
+# the agency name before the name can be used in a file name
+AGENCY_NAME_CHARS_TO_REPLACE = [".", " ", "/", "(", ")", "-", "&", "'"]
+
+
+class Command(BaseCommand):
+    help = "Assemble unlinked award by agency data into CSVs and Zip"
+    file_format = "csv"
+    filepaths_to_delete = []
+    total_download_count = 0
+    total_download_columns = 0
+    total_download_size = 0
+    working_dir_path = Path(settings.CSV_LOCAL_PATH)
+    full_timestamp = datetime.strftime(datetime.now(timezone.utc), "%Y-%m-%d_H%HM%MS%S%f")
+    _agency_name = None
+    _toptier_code = None
+
+    # KEY is the type of compute supported by this command
+    # key's VALUE are the strategies required by the compute type
+    # These strategies are used to change the behavior of this command
+    #   at runtime.
+    compute_types = {
+        ComputeTypeEnum.SPARK.value: {
+            "source_sql_strategy": {
+                "summary_file": summary_file,
+                "unlinked_contracts_file_d1": file_d1_sql_string,
+                "unlinked_assistance_file_d2": file_d2_sql_string,
+                "unlinked_accounts_file_c": file_c_sql_string,
+            },
+            "download_to_csv_strategy": SparkToCSVStrategy(logger=logger),
+            "readme_path": Path(settings.UNLINKED_AWARDS_DOWNLOAD_README_FILE_PATH),
+        },
+    }
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--compute-type",
+            choices=list(self.compute_types.keys()),
+            default=ComputeTypeEnum.SPARK.value,
+            help="Specify the type of compute to use when executing this command.",
+        )
+        parser.add_argument(
+            "--skip-upload",
+            action="store_true",
+            help="Don't store the list of IDs for downline ETL. Automatically skipped if --dry-run is provided",
+        )
+
+    def handle(self, *args, **options):
+        """
+        Generates a download data package specific to unlinked awards by agency
+        """
+        self.upload = not options["skip_upload"]
+        self.compute_type_arg = options.get("compute_type")
+        self.readme_path = self.compute_types[self.compute_type_arg]["readme_path"]
+
+        if self.compute_type_arg == ComputeTypeEnum.SPARK.value:
```

Comment:
If this is False, `self.spark` won't be defined. Later code in the handle method are dependent on this variable.

Reply:
Done